### ISSUE TITLE
Regression for initiating table column from list of structured arrays

### ISSUE THREAD

**Issue by mhvk** [astropy/astropy#9276]
This works on 3.2, but not on current master; the problem would seem that a structured data type is checked before any conversion from list to array is done. Likely introduced with #9048...
```
from astropy.table import Table
import numpy as np

Table([[np.array((1., 2.), 'f8,f8')] * 2])
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
/data/mhvk/venv/astropy-dev/lib/python3.7/site-packages/astropy-4.0.dev25759-py3.7-linux-x86_64.egg/astropy/table/table.py in _convert_data_to_col(self, data, copy, default_name, dtype, name)
    974             col = col_cls(name=name, data=data, dtype=dtype,
--> 975                           copy=copy, copy_indices=self._init_indices)
    976         except Exception:

/data/mhvk/venv/astropy-dev/lib/python3.7/site-packages/astropy-4.0.dev25759-py3.7-linux-x86_64.egg/astropy/table/column.py in __new__(cls, data, name, mask, fill_value, dtype, shape, length, description, unit, format, meta, copy, copy_indices)
   1229             # different dtype than self.
-> 1230             fill_value = self.dtype.type(data.fill_value)
   1231         self.fill_value = fill_value

ValueError: Can't cast from structure to non-structure, except if the structure only has a single field.

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
<ipython-input-10-3e6759525efb> in <module>()
----> 1 Table([[np.array((1., 2.), 'f8,f8')] * 2])

/data/mhvk/venv/astropy-dev/lib/python3.7/site-packages/astropy-4.0.dev25759-py3.7-linux-x86_64.egg/astropy/table/table.py in __init__(self, data, masked, names, dtype, meta, copy, rows, copy_indices, **kwargs)
    577 
    578         # Finally do the real initialization
--> 579         init_func(data, names, dtype, n_cols, copy)
    580 
    581         # Set table meta.  If copy=True then deepcopy meta otherwise use the

/data/mhvk/venv/astropy-dev/lib/python3.7/site-packages/astropy-4.0.dev25759-py3.7-linux-x86_64.egg/astropy/table/table.py in _init_from_list(self, data, names, dtype, n_cols, copy)
    841 
    842         for col, name, default_name, dtype in zip(data, names, default_names, dtype):
--> 843             col = self._convert_data_to_col(col, copy, default_name, dtype, name)
    844 
    845             cols.append(col)

/data/mhvk/venv/astropy-dev/lib/python3.7/site-packages/astropy-4.0.dev25759-py3.7-linux-x86_64.egg/astropy/table/table.py in _convert_data_to_col(self, data, copy, default_name, dtype, name)
    976         except Exception:
    977             # Broad exception class since we don't know what might go wrong
--> 978             raise ValueError('unable to convert data to Column for Table')
    979 
    980         col = self._convert_col_for_table(col)

ValueError: unable to convert data to Column for Table
```

**Comment by taldcroft** [comment]
Whack a mole...

**Comment by bsipocz** [comment]
I'm removing the milestone from issues, but any fix for this could still go into 4.0.

**Comment by bsipocz** [comment]
This has been fixed in master.
